packages/dd-trace/src/appsec/iast/analyzers/path-traversal-analyzer.js

```diff
@@ -0,0 +1,39 @@
+'use strict'
+const Analyzer = require('./vulnerability-analyzer')
+const { isTainted, getRanges } = require('../taint-tracking/operations')
+
+class PathTraversalAnalyzer extends Analyzer {
+  constructor () {
+    super('PATH_TRAVERSAL')
+    this.addSub('datadog:fs:access', path => this.analyze(path))
+    this.evidence = ''
+  }
+
+  _isVulnerable (pathArray, ctx) {
+    let ret = false
+    if (typeof pathArray !== 'object' || pathArray.constructor !== Array) {
+      return ret
+    }
+
+    for (const value of pathArray) {
+      if (typeof value === 'string') {
+        ret = isTainted(ctx, value)
+        if (ret) {
+          // Return the first vulnerable argument
+          this.evidence = value
```

Comment:
The analyzer should be stateless. How can we change this? Maybe we can change `Analyzer` class to work in different way if it has multiple values.